### PR TITLE
feat: Add pod identity association resource

### LINKED ISSUE #34561: [New]: EKS Pod Identity
### Description

AWS Announced new EKS Pod Identity feature, and AWS SDK Go v2 has been released including this feature.

### Requested Resource(s) and/or Data Source(s)

- `aws_eks_pod_identity_association`

```terraform
resource "aws_eks_pod_identity_association" "foo" {
  cluster_name = "my_cluster"
  namespace = "my_namespace"
  service_account_name = "foo-sa"
  iam_role_arn = "aws:......:foo"
}
```

### Potential Terraform Configuration

_No response_

### References

https://aws.amazon.com/ko/about-aws/whats-new/2023/11/amazon-eks-pod-identity/
https://aws.amazon.com/ko/blogs/aws/amazon-eks-pod-identity-simplifies-iam-permissions-for-applications-on-amazon-eks-clusters/
https://github.com/aws/aws-sdk-go-v2/blob/main/service/eks/CHANGELOG.md#v1340-2023-11-27
https://github.com/aws/aws-sdk-go-v2/blob/main/service/eksauth/CHANGELOG.md#v100-2023-11-27

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by bryantbiggs**
### Relations

- Closes #34561

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccEKSPodIdentityAssociation_ PKG=eks

...
```

### CODE REVIEW

**Comment by bryantbiggs** [comment]
@bschaatsbergen / @ewbankkit any ideas on whats missing for these errors? I'm still pretty new with the plugin framework 

```text
Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/eks/... -v -count 1 -parallel 20 -run='TestAccEKSPodIdentityAssociation_'  -timeout 360m
=== RUN   TestAccEKSPodIdentityAssociation_basic
=== PAUSE TestAccEKSPodIdentityAssociation_basic
=== RUN   TestAccEKSPodIdentityAssociation_disappears
=== PAUSE TestAccEKSPodIdentityAssociation_disappears
=== CONT  TestAccEKSPodIdentityAssociation_basic
=== CONT  TestAccEKSPodIdentityAssociation_disappears
=== NAME  TestAccEKSPodIdentityAssociation_basic
    pod_identity_association_test.go:33: Step 1/2 error: Error running apply: exit status 1
        
        Error: Provider returned invalid result object after apply
        
        After the apply operation, the provider still indicated an unknown value for
        aws_eks_pod_identity_association.test.modified_at. All values must be known
        after apply, so this is always a bug in the provider and should be reported
        in the provider's own repository. Terraform will still save the other known
        object values in the state.
=== NAME  TestAccEKSPodIdentityAssociation_disappears
    pod_identity_association_test.go:69: Step 1/1 error: Error running apply: exit status 1
        
        Error: Provider returned invalid result object after apply
        
        After the apply operation, the provider still indicated an unknown value for
        aws_eks_pod_identity_association.test.modified_at. All values must be known
        after apply, so this is always a bug in the provider and should be reported
        in the provider's own repository. Terraform will still save the other known
        object values in the state.
=== NAME  TestAccEKSPodIdentityAssociation_basic
    testing_new.go:90: Error running post-test destroy, there may be dangling resources: checking destroyed Amazon EKS (Elastic Kubernetes) Pod Identity Association (id-attribute-not-set): operation error EKS: DescribePodIdentityAssociation, https response error StatusCode: 400, RequestID: da2ac6a3-d821-4cc9-8426-8297d5427b03, InvalidParameterException: The parameter associationId should be 19 characters long.
--- FAIL: TestAccEKSPodIdentityAssociation_basic (676.56s)
=== NAME  TestAccEKSPodIdentityAssociation_disappears
    testing_new.go:90: Error running post-test destroy, there may be dangling resources: checking destroyed Amazon EKS (Elastic Kubernetes) Pod Identity Association (id-attribute-not-set): operation error EKS: DescribePodIdentityAssociation, https response error StatusCode: 400, RequestID: 0d7bcc59-7ab5-41c5-adf6-ba9b27fcfa42, InvalidParameterException: The parameter associationId should be 19 characters long.
--- FAIL: TestAccEKSPodIdentityAssociation_disappears (689.91s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/eks	690.065s
FAIL
make: *** [GNUmakefile:346: testacc] Error 1
```

**Comment by bryantbiggs** [comment]
Stuck on a state import issue - printing out the values, they look correct but its throwing a value conversion error on `tags_all`? (I think)

```text
make testacc TESTS=TestAccEKSPodIdentityAssociation_ PKG=eks
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/eks/... -v -count 1 -parallel 20 -run='TestAccEKSPodIdentityAssociation_'  -timeout 360m
=== RUN   TestAccEKSPodIdentityAssociation_basic
=== PAUSE TestAccEKSPodIdentityAssociation_basic
=== RUN   TestAccEKSPodIdentityAssociation_disappears
=== PAUSE TestAccEKSPodIdentityAssociation_disappears
=== CONT  TestAccEKSPodIdentityAssociation_basic
=== CONT  TestAccEKSPodIdentityAssociation_disappears
Import input: {ID:tf-acc-test-7003642029685638382:a-wqrjtcfptl1a6ze7a}
Import input ID: tf-acc-test-7003642029685638382:a-wqrjtcfptl1a6ze7a
Import cluster_name: tf-acc-test-7003642029685638382
Import association_id: a-wqrjtcfptl1a6ze7a
=== NAME  TestAccEKSPodIdentityAssociation_basic
    pod_identity_association_test.go:33: Step 2/2 error running import: exit status 1
        
        Error: Value Conversion Error
        
        An unexpected error was encountered while verifying an attribute value
        matched its expected type to prevent unexpected behavior or panics. This is
        always an error in the provider. Please report the following to the provider
        developer:
        
        Expected framework type from provider logic:
        types.MapType[basetypes.StringType] / underlying type:
        tftypes.Map[tftypes.String]
        Received framework type from provider logic: types.MapType[!!! MISSING TYPE
        !!!] / underlying type: tftypes.Map[tftypes.DynamicPseudoType]
        Path: tags_all
        
--- PASS: TestAccEKSPodIdentityAssociation_disappears (553.87s)
--- FAIL: TestAccEKSPodIdentityAssociation_basic (679.89s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/eks	680.030s
FAIL
make: *** [GNUmakefile:346: testacc] Error 1
```

**Comment by bryantbiggs** [comment]
```text
make testacc TESTS=TestAccEKSPodIdentityAssociation_ PKG=eks
==> Checking that code complies with gofmt requirements...
pick 14670e3cf7 fix: Remove tag manipulation logic
TF_ACC=1 go test ./internal/service/eks/... -v -count 1 -parallel 20 -run='TestAccEKSPodIdentityAssociation_'  -timeout 360m
=== RUN   TestAccEKSPodIdentityAssociation_basic
=== PAUSE TestAccEKSPodIdentityAssociation_basic
=== RUN   TestAccEKSPodIdentityAssociation_disappears
=== PAUSE TestAccEKSPodIdentityAssociation_disappears
=== CONT  TestAccEKSPodIdentityAssociation_basic
=== CONT  TestAccEKSPodIdentityAssociation_disappears
=== NAME  TestAccEKSPodIdentityAssociation_basic
    pod_identity_association_test.go:33: Step 1/2 error: Error running post-apply refresh: exit status 1

        Error: listing tags for EKS (Elastic Kubernetes) Pod Identity Association (a-62dlpjeabra3fxb2r)

          with aws_eks_pod_identity_association.test,
          on terraform_plugin_test.tf line 101, in resource "aws_eks_pod_identity_association" "test":
         101: resource "aws_eks_pod_identity_association" "test" {

        operation error EKS: ListTagsForResource, https response error StatusCode:
        400, RequestID: 3fcfb530-9365-4b34-9c7b-c9e1a81d0acf, BadRequestException:
        Invalid input resource arn: bad syntax
--- PASS: TestAccEKSPodIdentityAssociation_disappears (499.10s)
--- FAIL: TestAccEKSPodIdentityAssociation_basic (558.83s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/eks	561.939s
FAIL
make: *** [testacc] Error 1
```

File: internal/service/eks/pod_identity_association_test.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  name_prefix = %[1]q
+
+  assume_role_policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "eks.amazonaws.com"
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply internal/service/eks/pod_identity_association_test.go:173]: This is the IAM role for the cluster (control plane) - the Pod Identity role has the correct service principal (below) https://github.com/hashicorp/terraform-provider-aws/pull/34566/files#diff-cc8091fcb0bda70787e8bdd91679c19c12ec2eabd4176b47e31f21de62e163b1R238

File: internal/service/eks/pod_identity_association.go [hunk 2]
```diff
@@ -30,42 +30,43 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
-// Function annotations are used for resource registration to the Provider. DO NOT EDIT.
 // @FrameworkResource(name="Pod Identity Association")
 // @Tags(identifierAttribute="association_arn")
-func newResourcePodIdentityAssociation(_ context.Context) (resource.ResourceWithConfigure, error) {
-	r := &resourcePodIdentityAssociation{}
+func newPodIdentityAssociationResource(_ context.Context) (resource.ResourceWithConfigure, error) {
```
Comment (bryantbiggs) [inline internal/service/eks/pod_identity_association.go:35]: @ewbankkit these were generated by `skaff` - are there updates coming for it to use the recommended conventions?

Comment (ewbankkit) [reply internal/service/eks/pod_identity_association.go:35]: My personal taste 😄.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccEKSPodIdentityAssociation_' PKG=eks ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/eks/... -v -count 1 -parallel 2  -run=TestAccEKSPodIdentityAssociation_ -timeout 360m
=== RUN   TestAccEKSPodIdentityAssociation_basic
=== PAUSE TestAccEKSPodIdentityAssociation_basic
=== RUN   TestAccEKSPodIdentityAssociation_disappears
=== PAUSE TestAccEKSPodIdentityAssociation_disappears
=== RUN   TestAccEKSPodIdentityAssociation_tags
=== PAUSE TestAccEKSPodIdentityAssociation_tags
=== RUN   TestAccEKSPodIdentityAssociation_updateRoleARN
=== PAUSE TestAccEKSPodIdentityAssociation_updateRoleARN
=== CONT  TestAccEKSPodIdentityAssociation_basic
=== CONT  TestAccEKSPodIdentityAssociation_tags
--- PASS: TestAccEKSPodIdentityAssociation_basic (649.14s)
=== CONT  TestAccEKSPodIdentityAssociation_disappears
--- PASS: TestAccEKSPodIdentityAssociation_tags (693.67s)
=== CONT  TestAccEKSPodIdentityAssociation_updateRoleARN
--- PASS: TestAccEKSPodIdentityAssociation_disappears (603.78s)
--- PASS: TestAccEKSPodIdentityAssociation_updateRoleARN (654.07s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/eks	1349.519s
```

**Comment by ewbankkit** [comment]
@bryantbiggs Thanks for the contribution :tada: :clap:.

**Pull request merged**